src/PointwiseFunctions/MathFunctions/MathFunction.hpp

```diff
@@ -0,0 +1,90 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines base-class MathFunction.
+
+#pragma once
+
+#include <memory>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Options/Factory.hpp"
+
+/// Holds classes implementing MathFunction (functions \f$R\to R\f$).
+namespace MathFunctions {
+class Gaussian;
+class PowX;
+class Sinusoid;
+}  // namespace MathFunctions
+
+/*!
+ *  \ingroup MathFunctions
+ *  Encodes a function \f$R^n\to R\f$ where n is `VolumeDim`.
+ */
+template <size_t VolumeDim>
+class MathFunction : public Factory<MathFunction<VolumeDim>> {
+ public:
+  using creatable_classes = typelist<>;
+
+  MathFunction() = default;
+  MathFunction(const MathFunction& /*rhs*/) = delete;
+  MathFunction& operator=(const MathFunction& /*rhs*/) = delete;
+  MathFunction(MathFunction&& /*rhs*/) noexcept = default;             // NOLINT
+  MathFunction& operator=(MathFunction&& /*rhs*/) noexcept = default;  // NOLINT
+  virtual ~MathFunction() = default;
+
+  /// Returns the function value at the coordinate 'x'.
+  template <typename T, typename Fr>
+  T operator()(const tnsr::I<T, VolumeDim, Fr>& x) const;
```

Comment:
operator(), first_deriv, and second_deriv each take a different type as an argument.  That should be changed.